dev-tools/smoke/smoke_common.py

```
@@ -234,6 +242,10 @@ def wait_for_incremental_response(
             if not expects_explicit_failure and any(
                 stripped.startswith(prefix) for prefix in explicit_failure_prefixes
             ):
+                if stripped.startswith("ERROR UPSTREAM_FAILURE"):
+                    raise TransientUpstreamSmokeFailure(
+                        f"Command failed explicitly: {stripped}"
+                    )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)